staging/src/k8s.io/client-go/restmapper/shortcut.go

```diff
@@ -145,16 +148,33 @@ func (e shortcutExpander) expandResourceShortcut(resource schema.GroupVersionRes
 			}
 		}
 
+		found := false
+		var rsc schema.GroupVersionResource
 		for _, item := range shortcutResources {
 			if len(resource.Group) != 0 && resource.Group != item.ShortForm.Group {
 				continue
 			}
 			if resource.Resource == item.ShortForm.Resource {
-				resource.Resource = item.LongForm.Resource
-				resource.Group = item.LongForm.Group
-				return resource
+				if found {
+					if item.LongForm.Group == rsc.Group && item.LongForm.Resource == rsc.Resource {
+						// It is common and acceptable that group/resource has multiple
+						// versions registered in cluster. This does not introduce ambiguity
+						// in terms of shortname usage.
+						continue
+					}
+					if e.warningHandler != nil {
+						e.warningHandler(fmt.Sprintf("ambiguous resource %q for %s is skipped", item.LongForm, resource))
```

Comment:
I'd suggest something like this to read better:

```go
fmt.Sprintf("short name %q could also match lower priority resource %s.%s", resource.Resource, item.LongForm.Resource, item.LongForm.Group)
```

Reply:
Thanks for review.